### ISSUE TITLE
Bad results if charset is null in beginParsing

### ISSUE THREAD

**Issue by ludoo0d0a** [uniVocity/univocity-parsers#212]
I tried to guess delimiter .
If charset is missing or null, no default charset is internally provided and parsing goes wrong.
It returns ',' as delimiter (the default value I guess).

Here my simple test case :

```java
	@Test
	public void detectCsvFormatTst() throws IOException {
		String charsetName="UTF-8"; //ok
			charsetName="ISO-8859-1"; //ok
			//charsetName=null; //nok ; charset null

		String csvText = "a;b;c;d\n1;2;3;4";
		byte[] bytes = csvText.getBytes("UTF-8");
		ByteArrayInputStream bis = new ByteArrayInputStream(bytes);
		CsvParserSettings settings = new CsvParserSettings();
		settings.detectFormatAutomatically();
		CsvParser parser = new CsvParser(settings);
parser.beginParsing(bis, charsetName); //ok if charsetName not null
//parser.beginParsing(bis); //ko ; charset missing
		parser.beginParsing(bis);
		CsvFormat format = parser.getDetectedFormat();
		parser.stopParsing();
		assert(format.getDelimiter()==';');
	}
```

Please ensure null charset leads internally to a system default Charset.

**Comment by jbax** [comment]
Thanks for reporting this bug, I've fixed it and released a 2.6.0-SNAPSHOT build that includes the adjustment. Let me know how it goes for you.

I've got a few other bugs resolved already so I'll release a 2.5.9 build with this one too in a couple of days.

Thanks for using our parsers!

**Comment by ludoo0d0a** [comment]
Perfect ! thx

**Comment by jbax** [comment]
Released version 2.5.9 with this bugfix (and others)
